packages/dialog/test/dialog.test.js

```diff
@@ -89,28 +104,32 @@ describe('vaadin-dialog', () => {
     });
 
     describe('no-close-on-esc', () => {
-      it('should close itself on ESC press by default', () => {
+      it('should close itself on ESC press by default', async () => {
         esc(document.body);
-        expect(dialog.opened).to.eql(false);
+        await nextUpdate(dialog);
+        expect(dialog.opened).to.be.false;
       });
 
-      it('should not close itself on ESC press when no-close-on-esc is true', () => {
+      it('should not close itself on ESC press when no-close-on-esc is true', async () => {
         dialog.noCloseOnEsc = true;
+        await nextUpdate(dialog);
         esc(document.body);
-        expect(dialog.opened).to.eql(true);
+        expect(dialog.opened).to.be.true;
       });
     });
 
     describe('no-close-on-outside-click', () => {
-      it('should close itself on outside click by default', () => {
-        backdrop.dispatchEvent(new CustomEvent('click', { bubbles: true, composed: true }));
-        expect(dialog.opened).to.eql(false);
+      it('should close itself on outside click by default', async () => {
+        click(backdrop);
+        await nextUpdate(dialog);
+        expect(dialog.opened).to.be.false;
       });
 
-      it('should not close itself on outside click when no-close-on-outside-click is true', () => {
+      it('should not close itself on outside click when no-close-on-outside-click is true', async () => {
         dialog.noCloseOnOutsideClick = true;
-        backdrop.dispatchEvent(new CustomEvent('click', { bubbles: true, composed: true }));
-        expect(dialog.opened).to.eql(true);
+        await nextUpdate(dialog);
+        click(backdrop);
```

Comment:
Should we have a nextUpdate after click too?

```suggestion
        click(backdrop);
```

Reply:
Thanks, added.